earthaccess/store.py

```diff
@@ -200,40 +199,49 @@ def get_s3fs_session(
         Returns:
             a s3fs file instance
         """
-        if self.auth is not None:
-            if not any([concept_id, daac, provider, endpoint]):
-                raise ValueError(
-                    "At least one of the concept_id, daac, provider or endpoint"
-                    "parameters must be specified. "
-                )
+        if self.auth is None:
+            raise ValueError(
+                "A valid Earthdata login instance is required to retrieve S3 credentials"
+            )
+        if not any([concept_id, daac, provider, endpoint]):
+            raise ValueError(
+                "At least one of the concept_id, daac, provider or endpoint"
+                "parameters must be specified. "
+            )
+
+        # Get existing S3 credentials if we already have them
+        location = (concept_id, daac, provider, endpoint)
```

Comment:
Previously we were always returning the same `s3fs.S3FileSystem` instance, regardless of if the `provider=`, etc. has been changed. We now cache S3 credentials based on a unique `(concept_id, daac, provider, endpoint)` combination, which should be more correct.

Reply:
A brief comment about why all four of these things is needed would be useful here, I think. 

Naively, `concept_id`'s presence is a bit confusing to me since we're deriving the value of `provider` from it below (actually getting the credentials doesn't depend on `concept_id`). Could we reduce the size of the tuple (and elminate some of the if...elif chain below) if we did that derivation in advance if `provider is None`?